apis/r/src/Makevars.in

```diff
@@ -1,7 +1,9 @@
 CXX_STD = CXX17
 
-## We need the TileDB Headers, and for macOS aka Darwin need to set minimum version 10.14 for macOS
-PKG_CPPFLAGS = -I. -I../inst/include/ @tiledb_include@ @cxx17_macos@ -Iinclude
+## We need the TileDB Headers, and for macOS aka Darwin need to set minimum version 10.14 for macOS.
+## Our build assumes C++ libtiledbsoma in /usr/local/include and /usr/local/lib. Here we explicitly
+## add -I and -L for those, since while some compilers add these paths implicitly, others do not.
+PKG_CPPFLAGS = -Wno-deprecated-declarations -I. -I../inst/include/ -I/usr/local/include -L/usr/local/lib @tiledb_include@ @cxx17_macos@ -Iinclude
 
 ## We also need the TileDB library
 PKG_LIBS = @cxx17_macos@ -ltiledbsoma @tiledb_libs@ @tiledb_rpath@
```

Comment:
Note that we already query `pkg-config` from shell to fine `@tiledb_libs`.  We could check there for the library.

Reply:
Agreed that we want to use the `pkg-config` and hard coding `/usr/local` is not a good solution because this can very in every environment.

Reply:
And `pkg-config` is already used:

https://github.com/single-cell-data/TileDB-SOMA/blob/79511bb5929f33f0eff475a0ca88b92c04446121/apis/r/configure#L15-L33

Our problem may be the reliance on the `bld` script which works for us, but is not as general as we could make use of a proper artifact.

Reply:
@eddelbuettel @Shelnutt2 `/usr/local` was already hard-coded here:
https://github.com/single-cell-data/TileDB-SOMA/blob/1.2.3/.github/workflows/r-ci.yml#L41

The current R install assumes `/usr/local` in several places. We need to change this in all of those places.

Reply:
Also note `pkg-config` is a built-in on Linux, but on MacOS, it's just another optional package, installable via MacPorts or Homebrew. If we're going to take a hard dependency on it we need to enforce that hard dependency: put an install of `pkg-config` into our R CI; document this; and have `./configure` abort if `pkg-config` is not present.

Note in particular that this block
https://github.com/single-cell-data/TileDB-SOMA/blob/1.2.3/apis/r/configure#L17-L33
is never entered on vanilla MacOS install, because `pkg-config` doesn't exist on a vanilla MacOS install.

Alternatively, instead of taking a hard dependency on `pkg-config`, we can think of `pkg-config` as an optional dependency. We can _default_ to having our installer install `libtiledbsoma` into `/usr/local`. But allow the user to specify a different location.

Reply:
> @eddelbuettel @Shelnutt2 `/usr/local` was already hard-coded here: https://github.com/single-cell-data/TileDB-SOMA/blob/1.2.3/.github/workflows/r-ci.yml#L41
> 
> The current R install assumes `/usr/local` in several places.
> 
> If we do indeed to re-purpose / broaden this PR to change that pre-existing assumption, that's fine -- and we need to change it in all those places.

That is the installation prefix during CI. CI can configure the paths as we want them as its a runtime setting, based on where we install the package components. We can't rely on all environments being setup to use `/usr/local`. We should work to mimic the structure of the core TileDB library and TileDB-R, the package handling for finding a C++ library is a solved problem there and works well. Having the same approach here will keep it simpler and more supportable as we get more platforms, including windows support.

Reply:
@Shelnutt2 100% agreed on the end-goals -- we are on the same page